app/services/bulk_upload/sales/year2025/row_parser.rb

```diff
@@ -0,0 +1,2063 @@
+              category: :setup,
+            },
+            on: :after_log
+
+  validates :field_9,
+            inclusion: {
+              in: [2, 30, 18, 16, 24, 28, 31, 32],
+              if: proc { field_9.present? },
+              category: :setup,
+              question: QUESTIONS[:field_9].downcase,
+            },
+            on: :before_log
+
+  validates :field_9,
+            presence: {
+              message: I18n.t("#{ERROR_BASE_KEY}.not_answered", question: "type of shared ownership sale."),
+              category: :setup,
+              if: :shared_ownership?,
+            },
+            on: :after_log
+
+  validates :field_10,
+            inclusion: {
+              in: [1, 2],
+              if: proc { field_10.present? },
+              category: :setup,
+              question: QUESTIONS[:field_10].downcase,
+            },
+            on: :before_log
+
+  validates :field_10,
+            presence: {
+              message: I18n.t("#{ERROR_BASE_KEY}.not_answered", question: "staircasing transaction."),
+              category: :setup,
+              if: :shared_ownership?,
+            },
+            on: :after_log
+
+  validates :field_11,
+            inclusion: {
+              in: [8, 14, 27, 9, 29, 21, 22],
+              if: proc { field_11.present? },
+              category: :setup,
+              question: QUESTIONS[:field_11].downcase,
+            },
+            on: :before_log
+
+  validates :field_11,
+            presence: {
+              message: I18n.t("#{ERROR_BASE_KEY}.not_answered", question: "type of discounted ownership sale."),
+              category: :setup,
+              if: :discounted_ownership?,
+            },
+            on: :after_log
+
+  validates :field_115,
+            numericality: {
+              message: I18n.t("#{ERROR_BASE_KEY}.numeric.within_range", field: "Percentage discount", min: "0%", max: "70%"),
+              greater_than_or_equal_to: 0,
+              less_than_or_equal_to: 70,
+              if: :discounted_ownership?,
+            },
+            on: :before_log
+
+  validates :field_12,
+            presence: {
+              message: I18n.t("#{ERROR_BASE_KEY}.not_answered", question: "joint purchase."),
+              category: :setup,
+              if: :joint_purchase_asked?,
+            },
+            on: :after_log
+
+  validates :field_13,
+            presence: {
+              message: I18n.t("#{ERROR_BASE_KEY}.not_answered", question: "more than 2 joint buyers."),
+              category: :setup,
+              if: :joint_purchase?,
+            },
+            on: :after_log
+
+  validates :field_18,
+            presence: {
+              message: I18n.t("#{ERROR_BASE_KEY}.not_answered", question: "type of building."),
+              category: :setup,
+              if: :shared_or_discounted_but_not_staircasing?,
+            },
+            on: :after_log
```

Comment:
I don't think we need any of the new not answered validations here, cause they should be covered by `validate :validate_nulls`
We likely just want to keep the ones that we had, cause they're either in the setup section (`category: :setup`) or they have slightly different way of validating when coming through BU as opposed to validations we have on the log model.

Reply:
Yeah, I think it wasn't getting passed on from the model for some reason so it got added here in CLDC-3318, maybe there was another way to do it. We have other non setup validations here, but they're all defined as methods, we could extract this as a method, but I don't think it makes much of a difference